src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp

```diff
@@ -56,10 +58,41 @@ void EnemyBallPlacementPlay::ballPlacementWithShadow(
         tactics_to_run[0].emplace_back(crease_defenders[1]);
         tactics_to_run[0].emplace_back(crease_defenders[2]);
 
+        Vector ball_to_net =
+            (world.ball().position() - world.field().friendlyGoalCenter())
+                .normalize(-0.75 - ROBOT_MAX_RADIUS_METERS);
+
         Vector placement_to_net = (placement_point - world.field().friendlyGoalCenter())
                                       .normalize(-0.75 - ROBOT_MAX_RADIUS_METERS);
+
+        // If the enemy hasn't reached the ball yet, we use this flag to avoid shadowing
+        // so that we don't interfere with the enemy robots going to pick up the ball
+        if (!enemy_at_ball)
+        {
+            move_tactics[0]->updateControlParams(
+                world.ball().position() + ball_to_net +
+                    ball_to_net.perpendicular().normalize(1.25 * ROBOT_MAX_RADIUS_METERS),
+                ball_to_net.orientation() + Angle::half(), 0);
+            move_tactics[1]->updateControlParams(
+                world.ball().position() + ball_to_net -
+                    ball_to_net.perpendicular().normalize(1.25 * ROBOT_MAX_RADIUS_METERS),
+                ball_to_net.orientation() + Angle::half(), 0);
+            tactics_to_run[0].emplace_back(move_tactics[0]);
+            tactics_to_run[0].emplace_back(move_tactics[1]);
+
+            // Check to update flag
+            for (auto enemy_robot : world.enemyTeam().getAllRobotsExceptGoalie())
+            {
+                if (std::abs(
+                        (enemy_robot.position() - world.ball().position()).length()) <
+                    0.25)
+                {
+                    enemy_at_ball = true;
```

Comment:
I think it would make more sense to update `enemy_at_ball` prior to setting the move tactic (i.e. the `enemy_at_ball` is based on the most recent world, rather than the world from previous AI tick)
nit: we can also break after updating `enemy_at_ball`.

Reply:
bump

Reply:
fixed- good point

Reply:
It still looks to be taking an effect in the next iteration of the loop. The loop that updates `enemy_at_ball` should probably be moved to be before the main if statement on line 70 in a separate `if (!enemy_at_ball)` statement (there might be other possible approaches as well).

Reply:
ah yeah I see what I missed - its now updated before the if check